solutions/devsprint-cecilia-saraiva-2/FinanceAppTests/Service/FinanceServiceTests.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  FinanceServiceTests.swift
+//  FinanceAppTests
+//
+//  Created by Vitor Conceicao on 16/08/22.
+//
+
+import XCTest
+@testable import FinanceApp
+
+final class FinanceServiceTests: XCTestCase {
+    
+    func test_fetchHomeData_shouldnCallWithCorrectURLWhenCalled() {
+        let (sut, networkClient) = makeSut()
+        let stringURL = "https://raw.githubusercontent.com/devpass-tech/challenge-finance-app/main/api/home_endpoint.json"
+        
+        let expectation = XCTestExpectation(description: "Wait request")
+        
+        sut.fetchHomeData { _ in
+            XCTAssertTrue(networkClient.performRequestCalled)
+            XCTAssertEqual(networkClient.calledMethods, [.performRequest(url: stringURL)])
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 0.1)
+    }
+
+    func test_fetchHomeData_shouldReturnNilWhenRequestReturnNil() {
+        let (sut, networkClient) = makeSut()
+
+        let expectation = XCTestExpectation(description: "Wait request")
+        
+        sut.fetchHomeData { response in
+            XCTAssertTrue(networkClient.performRequestCalled)
+            XCTAssertNil(response)
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 0.1)
+    }
+    
+    func test_fetchHomeData_shouldReturnNilWhenRequestReturnEmptyData() {
+        let (sut, networkClient) = makeSut(data: Data())
+
+        let expectation = XCTestExpectation(description: "Wait request")
+        
+        sut.fetchHomeData { response in
+            XCTAssertTrue(networkClient.performRequestCalled)
+            XCTAssertNil(response)
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 0.1)
+    }
+    
+    func test_fetchHomeData_shouldReturnHomeDataWhenRequestReturnCorrectData() throws {
+        let activity = ActivityCodable(name: "Market", price: 10.0, time: "12:00:00")
+        let homeData = HomeDataCodable(balance: 100.0, savings: 200.0, spending: 150, activity: [activity])
+        let encodedHomeData = try XCTUnwrap(JSONEncoder().encode(homeData))
+        let (sut, networkClient) = makeSut(data: Data(encodedHomeData))
+
+        let expectation = XCTestExpectation(description: "Wait request")
+        
+        sut.fetchHomeData { response in
+            XCTAssertTrue(networkClient.performRequestCalled)
+            XCTAssertEqual(response?.balance, 100.0)
+            XCTAssertEqual(response?.activity.first?.name, "Market")
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 0.1)
+    }
+    
+    private func makeSut(data: Data? = nil) -> (FinanceService, NetworkClientSpy) {
```

Comment:
[sugestao] minor, como e um acronimo, acho que seria melhor:
```suggestion
    private func makeSUT(data: Data? = nil) -> (FinanceService, NetworkClientSpy) {
```

[sugestao] minor, Pra organizar melhor o arquivo talvez seja uma boa idea passar essa helper func para uma `private extension`. Assim temos os testes e os helpers separados. O que acha?

Reply:
A ideia aqui seria ter esse cara em cada arquivo de teste para preparar a SUT, então não teria como criar um helper generico onde todo mundo usa. Faz sentido?

Reply:
Acho que como escrevi nao ficou claro.
A minha primeira sugestao e quanto ao nome da func.
A segunda siugestao e colocar a func numa extensão privada no mesmo arquivo, nao em outro, com o objetivo de separar um pouco os testes com qualquer método helper nesse arquivo.
Espero que eu tenho escrito mais claro agora. 🙂

Reply:
Agora entendi Ceci, perdão, não tinha compreendido a primeira mensagem corretamente. Vou ajustar aqui